lib/finch/error.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Finch.Error do
+  @moduledoc """
+  An HTTP error.
+
+  This exception struct is used to represent errors of all sorts for
+  the HTTP/2 protocol.
+
+  A `Finch.Error` struct is an exception, so it can be raised as any other exception.
+
+  ## Message representation
+
+  If you want to convert an error reason to a human-friendly message (for example
+  for using in logs), you can use `Exception.message/1`:
+
+      iex> {:error, %Finch.Error{} = error} = request...
+      iex> Exception.message(error)
+
+  """
+
+  @type t() :: %__MODULE__{reason: atom()}
+
+  defexception [:reason]
+
+  @impl true
+  def exception(reason) do
+    %__MODULE__{reason: reason}
+  end
+
+  @impl true
+  def message(%__MODULE__{reason: reason}) when is_atom(reason) do
+    "error: #{reason}"
```

Comment:
nitpick: this is going to be logged as:

```
** (Finch.Error) error: connection_closed
```

and "just" error feels a bit redundant to me but it's pretty subjective. some other ideas:

1. `** (Finch.Error) connection_closed`
2. `** (Finch.Error) HTTP/2 error: connection_closed`
3. `** (Finch.HTTP2Error) connection_closed`

Reply:
Great, removing the string `error: `